packages/sanity/src/_internal/cli/server/__tests__/aliases.test.ts

```diff
@@ -0,0 +1,104 @@
+import path from 'node:path'
+
+import {describe, expect, it, jest} from '@jest/globals'
+import {escapeRegExp} from 'lodash'
+import resolve from 'resolve.exports'
+import {type Alias} from 'vite'
+
+import {browserCompatibleSanityPackageSpecifiers, getAliases} from '../aliases'
+
+const sanityPkgPath = path.resolve(__dirname, '../../../../../package.json')
+// eslint-disable-next-line import/no-dynamic-require
+const pkg = require(sanityPkgPath)
+
+describe('browserCompatibleSanityPackageSpecifiers', () => {
+  it('should have all specifiers listed in the package.json', () => {
+    const currentSpecifiers = Object.keys(pkg.exports)
+      .map((subpath) => path.join('sanity', subpath))
+      .sort()
+
+    // NOTE: this test is designed to fail if there are any changes to the
+    // package exports in the sanity package.json so you can stop and consider if that
+    // new subpath should also go into `browserCompatibleSanityPackageSpecifiers`.
+    // If there are changes, you may need to update this variable. New subpaths
+    // should go into `browserCompatibleSanityPackageSpecifiers` if that subpath
+    // is meant to be imported in the browser (e.g. a new subpath that is only meant
+    // for the CLI doesn't need to go into `browserCompatibleSanityPackageSpecifiers`).
+    expect(currentSpecifiers).toEqual([
+      'sanity',
+      'sanity/_internal',
+      'sanity/_singletons',
+      'sanity/cli',
+      'sanity/desk',
+      'sanity/migrate',
+      'sanity/package.json',
+      'sanity/presentation',
+      'sanity/router',
+      'sanity/structure',
+    ])
+
+    expect(browserCompatibleSanityPackageSpecifiers).toHaveLength(7)
+
+    for (const specifier of browserCompatibleSanityPackageSpecifiers) {
+      expect(currentSpecifiers).toContain(specifier)
+    }
+  })
+})
+
+describe('getAliases', () => {
```

Comment:
My personal preference is not to repeat the business logic and use shared utils in the test itself, but rather use something more aligned with a fixture (in this case, an array of `[{find: regex, replacement: path}]`) to ensure expected results and to make it a bit easier on a dev coming in trying to figure out what to expect.

With our dynamic builds and CI however, I'm not sure how feasible this is, so feel free to ignore this comment 😅

Reply:
![CleanShot 2024-07-18 at 19 28 05@2x](https://github.com/user-attachments/assets/d22237f2-38e9-424a-8b6d-21a29be6aa2b)
I tried to rewrite this test to be more of an integration test that actually calls vite.build but our jest environment is not having it. for now i think it's okay to merge this given i've tested it manually but i agree that this test could be improved. i've added a TODO comment for now